### DISCUSSION TITLE
It report errors when i use javascript mode. But when i use xml or json, it won't report error.

### DISCUSSION THREAD

**Discussion by darrenfei** [microsoft/monaco-editor#3758] [Q&A]
![1632989747533](https://user-images.githubusercontent.com/22699815/135414422-264a0c87-5427-43be-bfe1-1a447d06dbe2.jpg)
**monaco-editor version:** 0.28.1
**monaco-editor-webpack-plugin:**  4.2.0
**Browser:** Microsoft Edge 93.0.961.52
**OS:** macOs Big Sur 11.3.1
**Playground code that reproduces the issue:**

webpack plugin:
```
new MonacoWebpackPlugin({
      languages:[
        'xml',
        'json',
        'javascript'
      ],
      features:['coreCommands','find']
 })
```

code:
```
import { editor } from 'monaco-editor'

const codeString = 'console.log('code')'
editor.create(
      document.getElementById('editor'), 
      {
        value: codeString,
        theme: 'vs-dark',
        language: 'javascript',
        fontSize: 14,
        tabSize: 2,
        wordWrap: 'on'
      }
)
```

Error message below:
```
Could not create web worker(s). Falling back to loading web worker code in main thread, which might cause UI freezes. Please see https://github.com/microsoft/monaco-editor#faq
Unexpected token '<'
Unexpected usage
    at EditorSimpleWorker.loadForeignModule (editorSimpleWorker.js:450)
    at webWorker.js:38
```

**Comment by sentrychris** [comment]
I had a similar issue that was caused by not defining the location of my worker scripts. See https://github.com/microsoft/monaco-editor/blob/main/docs/integrate-esm.md for details on how to resolve.

**Comment by darrenfei** [comment]
> I had a similar issue that was caused by not defining the location of my worker scripts. See https://github.com/microsoft/monaco-editor/blob/main/docs/integrate-esm.md for details on how to resolve.

I tried, but failed.

**Comment by alexdima** [comment]
@darrenfei This looks like a problem with webpack. Could you please start with a sample like https://github.com/microsoft/monaco-editor-samples/tree/main/browser-esm-webpack-monaco-plugin and check if that works for you? Then you can try tweaking the webpack versions to be the same ones you have and check if things still work, then finally you can try to bring over the webpack config into your application's webpack config.

**Comment by darrenfei** [comment]
> @darrenfei This looks like a problem with webpack. Could you please start with a sample like https://github.com/microsoft/monaco-editor-samples/tree/main/browser-esm-webpack-monaco-plugin and check if that works for you? Then you can try tweaking the webpack versions to be the same ones you have and check if things still work, then finally you can try to bring over the webpack config into your application's webpack config.

I created a new react app with create-react-app & react-app-rewired. It still not working.
Here is the config:
![image](https://user-images.githubusercontent.com/22699815/136512213-2d2fa2d0-1ea8-4924-9a97-a33444489391.png)
![image](https://user-images.githubusercontent.com/22699815/136512259-214c54cd-6c69-4f8e-ae2f-3b7f63447ab6.png)
![image](https://user-images.githubusercontent.com/22699815/136512280-0517c742-56f9-4a7f-b79f-5b47b3e9ad12.png)

**Reply by fantasy525** [reply]
I solved it，in fact,if you use javascript,u can set 
`  config
      .plugin('monaco-editor-webpack-plugin')
      .use(monacoEditorPlugin, [{ languages: ['typescript'] }]);`
and the default language for monaco-editor is typescript ,html,json,css
<img width="261" alt="image" src="https://github.com/microsoft/monaco-editor/assets/22516157/6e024e64-a542-4712-b0cf-cd756bbbe8b5">
and I change  languages:['javascript'] to languages:['typesccript'],then OK
